tools/clang/test/CodeGenSPIRV/fn.param.inout.type-mismatch.hlsl

```diff
@@ -17,7 +23,17 @@ float4 main() : SV_Target0 {
 // CHECK-NEXT: [[outputFloat3_0:%[0-9]+]] = OpFConvert %v3float [[outputHalf3_0]]
 // CHECK-NEXT:                         OpStore %output [[outputFloat3_0]]
 
-// CHECK-NEXT: [[outputFloat3_1:%[0-9]+]] = OpLoad %v3float %output
+// CHECK:      [[f:%[0-9]+]] = OpLoad %float %f
+// CHECK-NEXT: [[splat:%[0-9]+]] = OpCompositeConstruct %v3float [[f]] [[f]] [[f]]
+// CHECK-NEXT:      OpStore %param_var_p [[splat]]
+// CHECK-NEXT: OpFunctionCall %void %bar %param_var_p
+// CHECK-NEXT: [[ret:%[0-9]+]] = OpLoad %v3float %param_var_p
+// CHECK-NEXT: [[ext:%[0-9]+]] = OpCompositeExtract %float [[ret]] 0
+// CHECK-NEXT:      OpStore %f [[ext]]
+   float f = 0;
+   bar(f);
+
```

Comment:
The only surprise is the f1 case. The f3 case has the types matching, so naturally that works. I has tested f2 and f4 earlier, and they fail in sema for both DXIL and SPIRV. So that is fine.

I did not think about f1. The DXIL path fails as well, and it probably fail in sema. We can follow up with a new issue for both SPIRV and DXIL.